Respect `with_locals` setting

alxwrd opened the pull request:
Hi!

This is a patch to use the `with_locals` setting from the Sentry client when creating an event.

[`with_locals`](https://docs.sentry.io/platforms/python/configuration/options/#with-locals) is a setting you can enable when initialising the Sentry SDK (

```python
sentry_sdk.init(
    with_locals=False,
    ...
)
```

) that can turn off local varibales being displayed with each stackframe.

[`event_from_exception`](https://github.com/getsentry/sentry-python/blob/8b1e8ce5f69265016ccc640b86ea1573749e23aa/sentry_sdk/utils.py#L739) uses it's own "`client_options`" which is used when [creating the frames](https://github.com/getsentry/sentry-python/blob/8b1e8ce5f69265016ccc640b86ea1573749e23aa/sentry_sdk/utils.py#L562-L570).

alxwrd commented:
- https://github.com/kiwicom/structlog-sentry/pull/74

alxwrd commented:
Current work around

```python
import functools

import sentry_sdk
import structlog
import structlog_sentry

structlog_sentry.event_from_exception = functools.partial(
    structlog_sentry.event_from_exception,
    client_options={
        "with_locals": sentry_sdk.Hub.current.client.options.get("with_locals")
    },
)

structlog.configure(
    processors=[structlog_sentry.SentryProcessor],
)
```

paveldedik commented:
Hi @alxwrd Thank you for the PR, could you please rebase your branch onto master? There are going to be some changes (you will definitely need to update tests).

For all changes see https://github.com/kiwicom/structlog-sentry/pull/79

alexward-plutoflume commented:
hi @paveldedik, thank you. That's done :)

alxwrd commented:
hi @paveldedik, thank you. That's done :)

paveldedik commented on test/test_sentry_processor.py:
```diff
@@ -236,3 +237,41 @@ def test_sentry_ignore_logger(sentry_events, level):
+
+
+def test_sentry_json_respects_global_with_locals_option_no_locals():
+    with unittest.mock.patch.object(SentryProcessor, "_get_hub") as mock__get_hub:
```
Could you please use `sentry_events` fixture instead? It does not require mocking and it contains captured sentry events. For inspiration see other tests.

paveldedik commented on test/test_sentry_processor.py:
```diff
@@ -236,3 +237,41 @@ def test_sentry_ignore_logger(sentry_events, level):
+
+
+def test_sentry_json_respects_global_with_locals_option_with_locals():
+    with unittest.mock.patch.object(SentryProcessor, "_get_hub") as mock__get_hub:
```
The same as above.

paveldedik reviewed (changes requested):
Thank you for the PR, it looks good :-), I have just a minor thing.

Also please rebase again (sorry for that, the CI weren't running for your branch so I fixed that in master).

alxwrd commented:
hi @paveldedik - I've updated the tests and rebased.

To pass the configuration into `sentry_sdk.Client`, I've used `@pytest.mark.parametrize` with a notion of `ClientParams`. If you'd rather this doing a different way, let me know.

paveldedik reviewed (approved):
Thank you! I like this approach :-)
